Minio.Functional.Tests/FunctionalTest.cs

```diff
@@ -752,7 +751,6 @@ private static void runQuickTests(MinioClient minioClient)
             }
             catch (Exception ex)
             {
-                Assert.Fail();
                 throw ex;
             }
             return;
```

Comment:
My question is similar to @ebozduman's - doesn't throwing and catching an exception (especially of the base `Exception` type) have absolutely no effect? I am not too familiar with C#, but to illustrate my point in Python:

```py
def g():
    try:
        return f()
    except Exception as e:
        raise e

# Now calling f() and g() have exactly the same behaviour in terms 
# of values returned and exceptions thrown.
```

If this this the case, then all blocks like:

```
catch (Exception ex)
{
    throw ex
}
````
may be removed and it will have no effect on the program behaviour.

Reply:
except in cases where we are intercepting OperationCanceledException, removed the unnecessary rethrowing of error.